### PR TITLE
Partial fix for Teeter Dance and Ability Dancer interaction

### LINKED ISSUE #4101: Dancer + Teeter Dance
### Description

Dancer is supposed to copy Teeter Dance (which is listed as a danceMove). However, it does not work. Perhaps all dance moves with the Flag `MOVE_TARGET_FOES_AND_ALLY` fail to get copied.

### Version

upcoming (Edge)

### Upcoming/master Version

09d12fb

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Teeter Dance used a new script for the confusion effect but `EFFECT_CONFUSE` can be reused since everything else will be handled by the target flag. This partially fixes #4101. Meaning it fixes the interaction in single battles. In doubles Dancer copies the move but confuses only one opposing pokemon.

The problem in double battles might be a larger issue were properties of the called move aren't copied. This has been an issue before with Metronome and Mirror Move.

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Closing this for now. Seems like this was the wrong approach

**Comment by AlexOn1ine** [comment]
Sorry, I was really sloppy yesterday. Should be good for review now.

File: test/battle/ability/dancer.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include "global.h"
+#include "test/battle.h"
+
+SINGLE_BATTLE_TEST("Dancer can copied a dance move immediately after it was used addionally to the users move")
```
Comment (Bassoonian) [inline test/battle/ability/dancer.c:4]:
```suggestion
SINGLE_BATTLE_TEST("Dancer can copy a dance move immediately after it was used addionally to the users move")
```

**Review by Bassoonian — CHANGES_REQUESTED** [review]

File: test/battle/ability/dancer.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#include "global.h"
+#include "test/battle.h"
+
+SINGLE_BATTLE_TEST("Dancer can copy a dance move immediately after it was used addionally to the users move")
```
Comment (ultima-soul) [inline test/battle/ability/dancer.c:4]:
```suggestion
SINGLE_BATTLE_TEST("Dancer can copy a dance move immediately after it was used and allow the user of Dancer to still use its move")
```
Does this capture the intent of test better?

Comment (AlexOn1ine) [reply test/battle/ability/dancer.c:4]: it does, ye

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**